RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -305,214 +471,193 @@ void EvtPlaneProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSetup
         }
       }
     }
-
-    //
-    //Get flattening parameter file.
-    //
-    if (loadDB_) {
-      edm::ESHandle<RPFlatParams> flatparmsDB_;
-      iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
-      LoadEPDB db(flatparmsDB_, flat);
-      if (!db.IsSuccess()) {
-        loadDB_ = kFALSE;
-      }
+  }
+  //
+  //Get flattening parameter file.
+  //
+  if (loadDB_ && hirpWatcher.check(iSetup)) {
+    edm::ESHandle<RPFlatParams> flatparmsDB_;
+    iSetup.get<HeavyIonRPRcd>().get(flatparmsDB_);
+    LoadEPDB db(flatparmsDB_, flat);
+    if (!db.IsSuccess()) {
+      loadDB_ = kFALSE;
     }
-
-  }  //rp record change
-
+  }
   //
   //Get Centrality
   //
   int bin = 0;
+  int cbin = 0;
   if (loadDB_) {
     edm::Handle<int> cbin_;
     iEvent.getByToken(centralityBinToken, cbin_);
-    int cbin = *cbin_;
+    cbin = *cbin_;
```

Comment:
```suggestion
cbin = iEvent.get(centralityBinToken);
```
is a more recently available shorter interface.

Is this really supposed to be done only when loadDB_ is false ?